src/common/utility_mac.mm

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (C) by Daniel Molkentin <danimo@owncloud.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "result.h"
+#include "utility.h"
+
+#include <QCoreApplication>
+#include <QDir>
+#include <QtMacExtras/QtMacExtras>
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <CoreServices/CoreServices.h>
+#import <Foundation/NSFileManager.h>
+
+namespace OCC {
+
+void setupFavLink_private(const QString &folder)
+{
+    // Finder: Place under "Places"/"Favorites" on the left sidebar
+    CFStringRef folderCFStr = CFStringCreateWithCString(0, folder.toUtf8().data(), kCFStringEncodingUTF8);
+    CFURLRef urlRef = CFURLCreateWithFileSystemPath(0, folderCFStr, kCFURLPOSIXPathStyle, true);
+
+    LSSharedFileListRef placesItems = LSSharedFileListCreate(0, kLSSharedFileListFavoriteItems, 0);
+    if (placesItems) {
+        //Insert an item to the list.
+        LSSharedFileListItemRef item = LSSharedFileListInsertItemURL(placesItems,
+            kLSSharedFileListItemLast, 0, 0,
+            urlRef, 0, 0);
+        if (item)
+            CFRelease(item);
+    }
+    CFRelease(placesItems);
+    CFRelease(folderCFStr);
+    CFRelease(urlRef);
+}
+
+
+static Result<void, QString> writePlistToFile(NSString *plistFile, NSDictionary *plist)
+{
+    // Our deployment target is 10.12, but in the 10.13 SDK `writeToFile:atomically:` is deprecated. So let's future-proof this code:
+    if (@available(macOS 10.13, *)) {
+        NSError *error = nil;
+        if (![plist writeToURL:[NSURL fileURLWithPath:plistFile isDirectory:NO] error:&error]) {
+            return QString::fromNSString(error.localizedDescription);
+        }
+    } else {
```

Comment:
Maybe we should target 2.10 with this one, it would remove the requirement so support < 10.13

You could also switch to std::optional https://github.com/owncloud/client/pull/9107

Reply:
done